src/vault.py

```diff
@@ -319,29 +284,29 @@ def _generate_root_certificate(self, ttl: str) -> str:
             "common_name": ("Vault Root Certificate Authority " "({})".format(CHARM_PKI_MP)),
             "ttl": ttl,
         }
-        root_certificate = self.client.write(
+        root_certificate = self._client.write(
             "{}/root/generate/internal".format(CHARM_PKI_MP), **config
         )
         if not root_certificate["data"]:
             raise Exception(root_certificate.get("warnings", "unknown error"))
         return root_certificate["data"]["certificate"]
 
-    def _enable_secrets_engine(self, name: str, ttl: str = None, max_ttl: str = None) -> None:
+    def enable_secrets_engine(self, ttl: str = None, max_ttl: str = None) -> None:
         """Enables Vault's secret engine if the backend is mounted.
 
         Args:
-            name (str): Name of backend to enable.
             ttl (str): Time to live.
+            max_ttl (str): Max Time to live.
 
         Returns:
             None
         """
-        logger.info("Vault - Enabling secrets engine")
-        if not self._is_backend_mounted(name):
-            self.client.sys.enable_secrets_engine(
+        if not self._is_backend_mounted(CHARM_PKI_MP):
+            logger.info("Vault - Enabling secrets engine")
```

Comment:
A general comment, I think you can drop `Vault -` from all your log messages - the juju uniter will already include the unit name in each log line. In this case I'd probably go for something like:

```
logger.info("Enabled PKI secrets engine")
```

After the call to `self._client.sys.enable_secrets_engine(...)`

In general, prefer logging when something *has been done* over when you're starting something, or the logs can be a little confusing when the subsequent operation fails for whatever reason :)

Reply:
Done